packages/api/lib/writeRecords/write-granules.js

```diff
@@ -606,26 +707,27 @@ const _writeGranule = async ({
     postgresGranuleRecord,
     writeConstraints,
   });
+  const pgGranule = writePgGranuleResult.pgGranule;
 
   // Files are only written to Postgres if the granule is in a "final" state
   // (e.g. "status: completed") and there is a valid `files` key in the granule.
   // An empty array of files will remove existing file records but a missing
   // `files` key will not.
-  if ((writeConstraints === false || (isStatusFinalState(status))) && 'files' in apiGranuleRecord) {
-    await _writeGranuleFiles({
-      granuleCumulusId: pgGranule.cumulus_id,
-      granule: apiGranuleRecord,
-      knex,
+  if (writePgGranuleResult.status === 'success') {
```

Comment:
We should reverse this if and the comment.

Reply:
sorry for being a little dense - do you mean to change it to `if (!(writePgGranuleResult.status !== 'success'))` or `if(writePgGranuleResult.status !== 'dropped')`? Also, what do you mean by reversing the comment?

Reply:
Resolved in [0ad1654](https://github.com/nasa/cumulus/pull/3242/commits/0ad165404de0e900c2b7b1ae7be080e5f61427c3)